CONTRIBUTING.md

```diff
@@ -1,42 +1,172 @@
-# How to contribute
+# Contribution Guidelines
+
+* [Prerequisites](#prerequisites)
+* [Definition of trivial contributions](#definition-of-trivial-contributions)
+* [Code](#code)
+  * [Code style](#code-style)
+  * [Dependencies](#dependencies)
+  * [Unit tests](#unit-tests)
+* [Contributing process](#contributing-process)
+  * [Get buyoff or find open community issues or features](#get-buyoff-or-find-open-community-issues-or-features)
+  * [Set up your environment](#Set-up-your-environment)
+  * [Prepare commits](#prepare-commits)
+  * [Submit pull request](#Submit-pull-request)
+  * [Respond to feedback on pull request](#respond-to-feedback-on-pull-request)
+* [Other general information](#other-general-information)
+* [Acknowledgement](#acknowledgement)
+
+## Prerequisites
 
-So you're thinking about contributing to Cake? Great! **It's really appreciated.**
+By contributing to Cake, you assert that:
 
-## Where to start?
+* The contribution is your own original work.
+* You have the right to assign the copyright for the work (it is not owned by your employer, or
+  you have been given copyright assignment in writing).
+* You [license](https://github.com/cake-build/cake/blob/develop/LICENSE) the contribution under the terms applied to the rest of the Cake project.
+* You agree to follow the [code of conduct](https://github.com/cake-build/cake/blob/develop/CODEOFCONDUCT.md).
 
-Start by either suggest a feature in the issue list that you want to see in Cake, or assign yourself to an existing feature/bug. Make sure that no one else is working on it already though. It's also appreciated to have some kind of discussion about the issue if it's not an easy fix.
+## Definition of trivial contributions
+It's hard to define what is a trivial contribution. Sometimes even a 1 character change can be considered significant.
+Unfortunately because it can be subjective, the decision on what is trivial comes from the committers of the project
```

Comment:
@daveaglick I agree with this.  @devlead can you fix up, and I will pull in, or if you are not in a position to do that, I can fix after merging.